apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx

```diff
@@ -29,16 +31,34 @@ export function ImportLedgerAccounts() {
     const accountsUrl = useNextMenuUrl(true, `/accounts`);
     const navigate = useNavigate();
 
-    const onDeriveError = useCallback(() => {
-        navigate(accountsUrl, { replace: true });
-        toast.error('Make sure you have the Sui application open.');
-    }, [accountsUrl, navigate]);
+    const existingAccounts = useAccounts();
+    const [selectedLedgerAccounts, setSelectedLedgerAccounts] = useState<
+        SerializedLedgerAccount[]
+    >([]);
 
-    const [ledgerAccounts, setLedgerAccounts, areLedgerAccountsLoading] =
-        useDeriveLedgerAccounts({
-            numAccountsToDerive: numLedgerAccountsToDeriveByDefault,
-            onError: onDeriveError,
-        });
+    const {
+        data: ledgerAccounts,
+        isLoading: areLedgerAccountsLoading,
+        isError: encounteredDerviceAccountsError,
+    } = useDeriveLedgerAccounts({
+        numAccountsToDerive: numLedgerAccountsToDeriveByDefault,
+        select: (ledgerAccounts) => {
+            const existingAccountAddresses = existingAccounts.map(
+                (account) => account.address
+            );
+            return ledgerAccounts.filter(({ address }) => {
+                return !existingAccountAddresses.includes(address);
+            });
+        },
+        onError: (error) => {
+            if (error instanceof LockedDeviceError) {
+                toast.error('Your device is locked. Un-lock it and try again.');
+            } else {
+                toast.error('Make sure you have the Sui application open.');
```

Comment:
`Make sure you have the Sui application on your Ledger device` to be a bit more verbose?